report/sections/Chapter 04 - Design/design.tex

```diff
@@ -0,0 +1,41 @@
+
+\section{Wireframe for browsing}
+The design for the page displayed in figure \ref{fig:wfExercise} does not satisfy the entirety of the use case \textbf{Solve exercises}, as we have designed a separate page to browse exercises as well as sessions and syllabi. The page structure for all of these are similar, and therefore we only show an example of a page for browsing syllabi.
+% Structure of creating sessions and syllabi
+\begin{figure}[H]
+    \includegraphics[scale=0.3]{browseSyllabus.png}
+    \centering
+    \caption{wireframe for browsing a syllabus}
+    \label{fig:wfSyllabus}
+\end{figure}
+Note that the buttons related to creating new items or updating items in the list will only be available to the lecturer. The items shown in the list corresponds to which page that the user is on. If the user is browsing syllabi, then the items contained in the list will be syllabi. If the user is browsing sessions, the list will contain sessions. Similarly for exercises. This design satisfies the section of the use case \textbf{Solve exercises}, related to choosing sessions and exercises. Additionally, it also satisfies part of the use case \textbf{Create exercises} related to having the ability to create and edit syllabi and exercises.
+
+\section{Wireframe for creating an exercise}
+The wireframe in figure \ref{fig:wfProblem}, shows the page that the lecturer will see, when the create exercise button when browsing exercises. 
+% create exercises
+\begin{figure}[H]
+	\includegraphics[scale=0.6]{createProblem.jpg}
+	\centering
+	\caption{wireframe for creating a problem}
+	\label{fig:wfProblem}
+\end{figure}
+
+This page allows the lecturer to create an exercise with a title, test, function template and description. These are fields required to provide the information necessary for a student to solve the exercises. Additionally, test field allows a lecturer to specify the tests that student's code must pass. The function template field allows the lecturer to specify functions that the test will call, and which will be included in the editor on the solve exercise page. This satisfies the remaining requirements of the use case \textbf{Create exercises}.
+
+As mentioned previously not all the use cases have been designed for, because the remaining use cases and related features were deemed as nonessential for a minimal viable product. The next chapter will cover the system architecture.
```

Comment:
I just noticed that the MVP section isn't actually included in our report so far. That is, `MVP.tex` isn't included. That makes this the first place we mention a minimal viable product. Not sure if that's great for the reader---they won't know what that is.